### ISSUE TITLE
states.ssh_auth adds bogus newline before newly added keys

### ISSUE THREAD

**Issue by mstarostik** [saltstack/salt#33793]
### Description of Issue/Question

The ssh_auth state module adds a newline before every new key it adds to the authorized_keys file.

Problem is a wrong seek() call in ssh.set_auth_key(), modules/ssh.py line 706 (676 in develop branch).  If I change _fh.seek(1024, 2) to _fh.seek(-1, 2), the issue goes away.  But that can't be the real fix as it breaks (invalid argument exception) if for whatever reason, authorized_keys is zero bytes long.
### Setup

Anything along the lines of:

```
test:
  ssh_auth.present:
    - user: test
    - names:
      - BOGUSKEY
```
### Steps to Reproduce Issue

Create a state like above and apply, making sure the key is not present before.
### Versions Report

```
Salt Version:
           Salt: 2015.8.8

Dependency Versions:
         Jinja2: 2.8
       M2Crypto: Not Installed
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.1
         Python: 2.7.10 (default, May 12 2016, 18:33:09)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.1
           cffi: 1.2.1
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.5
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: gentoo 2.2 
        machine: x86_64
        release: 4.4.6-gentoo-jmb
         system: Gentoo Base System 2.2 
```

**Comment by Ch3LL** [comment]
@mstarostik I am able to replicate this behavior on 2016.3.0 and 2015.5.9 as well with the example in the docs:

```
sshkeys:
  ssh_auth.present:
    - user: root
    - enc: ssh-rsa
    - options:
      - option1="value1"
      - option2="value2 flag2"
    - comment: myuser
    - names:
      - AAAAB3NzaC1kc3MAAACBAL0sQ9fJ5bYTEyY==
      - ssh-dss AAAAB3NzaCL0sQ9fJ5bYTEyY== user@domain
      - option3="value3" ssh-dss AAAAB3NzaC1kcQ9J5bYTEyY== other@testdomain
      - AAAAB3NzaC1kcQ9fJFF435bYTEyY== newcomment
```

```
➜  salt sudo cat /root/.ssh/authorized_keys

option1="value1",option2="value2 flag2" ssh-rsa AAAAB3NzaC1kcQ9fJFF435bYTEyY== newcomment

option1="value1",option2="value2 flag2" ssh-dss AAAAB3NzaCL0sQ9fJ5bYTEyY== user@domain

option3="value3" ssh-dss AAAAB3NzaC1kcQ9J5bYTEyY== other@testdomain

option1="value1",option2="value2 flag2" ssh-rsa AAAAB3NzaC1kc3MAAACBAL0sQ9fJ5bYTEyY== myuser
```

**Comment by rallytime** [comment]
Closed via #43483
